### PR TITLE
Only show customers that are waiting for invoices

### PR DESCRIPTION
**Pull request by mayabrandi**
Making a customers_to_invoice list to only show customers that are waiting for invoices. 

Also taking away requirement received_at - Samples that should be invoiced dont need this requirement anymore

### CODE REVIEW

**Comment by mayabrandi** [comment]
Introduced a bug! Wait with merging till Ive fixed it!

**Comment by ingkebil** [comment]
You're working on a package that is not owned by you. As it says above the "Merge pull request" button: "At least 1 approving review is required by reviewers with write access."

File: cg/server/invoices/templates/invoices/index.html [hunk 1]
```diff
@@ -61,8 +61,8 @@ <h1>Invoices</h1>
       </a>
       </td>
       <td>{{ invoice.customer.internal_id}} - {{ invoice.customer.name }}</td>
-      <td>{{ invoice.created_at.strftime('%Y-%m-%d') }}</td>
```
Comment (mayabrandi) [inline cg/server/invoices/templates/invoices/index.html]: adding a exception here.  Sort of a bug-fix

File: cg/server/invoices/templates/invoices/invoice.html [hunk 2]
```diff
@@ -98,12 +98,24 @@ <h1>Invoice template</h1>
...
+                    <div class="col-sm-9">
+                    <button class="btn btn-primary" type="submit">Undo Invoice</button> 
+                    <input type="hidden" name="samples" value={{ invoice_dict.records }}>
+                    <input type="hidden" name="invoice_id" value={{ invoice.id }}>
+                    <input type="hidden" name="record_type" value={{ record_type }}>
+                    <input type="hidden" name="undo" value=1>
+                  </div>
+                  </div>
+                  </form> 
                   </div>
```
Comment (mayabrandi) [inline cg/server/invoices/templates/invoices/invoice.html]: Adding a new form to be able to undo a invoice.

File: cg/server/invoices/templates/invoices/invoice.html [hunk 3]
```diff
@@ -166,7 +178,7 @@ <h5 align="right">Fakturanummer: {{ invoice.id }}</h5>
                 <td>{{record.lims_id}}</td>
                 <td>{{record.application_tag}}</td>
                 <td>{{record.project}}</td>
-                <td>{{record.date.date()}}</td>
+                <td>{{record.date}}</td>
                 <td>{{record.price}}</td>
```
Comment (mayabrandi) [inline cg/server/invoices/templates/invoices/invoice.html]: bug-fix

File: cg/server/invoices/views.py [hunk 4]
```diff
@@ -20,7 +20,22 @@ def index():
...
+            if record_type=='Pool':
+                pool=db.pool(record.id)
+                pool.delete('invoice_id')
+            elif record_type=='Sample':
+                sample=db.sample(record.id)
+                sample.delete('invoice_id')    
+        db.session.commit()
+        return redirect(url_for('.new', record_type=record_type))
+    elif request.method == 'POST':        
         customer_id = request.form.get('customer')
```
Comment (mayabrandi) [inline cg/server/invoices/views.py]: deleting the invoice object and the invoice ids from the sample/pool-objects

File: cg/server/invoices/views.py [hunk 5]
```diff
@@ -116,7 +137,8 @@ def invoice(invoice_id):
                             invoice = invoice_obj, 
                             invoice_dict = invoice_dict,
                             total_price = total_price,
-                            final_price = final_price)
+                            final_price = final_price,
+                            record_type=record_type)
```
Comment (mayabrandi) [inline cg/server/invoices/views.py]: record-type is needed for the backing

File: cg/store/models.py [hunk 6]
```diff
@@ -420,6 +420,7 @@ class Invoice(Model):
     excel_kth = Column(types.BLOB)
     excel_ki = Column(types.BLOB)
     price = Column(types.Integer)
+    record_type = Column(types.Text)
```
Comment (mayabrandi) [inline cg/store/models.py]: I added record_type to the invoice-table in the cg-database. Reflected here in the invice model

**Comment by mayabrandi** [comment]
@ingkebil This is now ready to be merged from my point of view. Its been tested locally against the cg-stage database. Added some comments in the PR.

**Pull request merged**